sdk/src/sysvars/instructions.rs

```diff
@@ -133,10 +130,34 @@ impl<'a> TryFrom<&'a AccountView> for Instructions<Ref<'a, [u8]>> {
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct IntrospectedInstruction<'a> {
     pub raw: *const u8,
-    pub marker: PhantomData<&'a [u8]>,
+    marker: PhantomData<&'a [u8]>,
 }
 
 impl IntrospectedInstruction<'_> {
+    /// Create a new IntrospectedInstruction.
+    ///
+    /// # Safety
+    ///
+    /// This function is unsafe because it does not verify anything about the pointer.
+    ///
+    /// It is private and used internally within the `get_instruction_account_at` function, which
+    /// performs the necessary index verification. However, to optimize performance for users
+    /// who are sure that the index is in bounds, we have exposed it as an unsafe function.
+    #[inline(always)]
+    unsafe fn new_unchecked(raw: *const u8) -> Self {
+        Self {
+            raw,
+            marker: PhantomData,
+        }
+    }
+
+    /// Load the number of instructions in the currently executing `Transaction`.
```

Comment:
Here it should be `Instruction` right? Something like:
```suggestion
    /// Get the number of accounts of the `Instruction`.
```